# Enables the --enable-async-dns feature on all platforms
# The async DNS is Chromium's built-in DNS client. It is needed to enable
# DNS over HTTPS.

--- a/chrome/browser/about_flags.cc
+++ b/chrome/browser/about_flags.cc
@@ -3537,11 +3537,9 @@ const FeatureEntry kFeatureEntries[] = {
                                     kSpeculativePreconnectFeatureVariations,
                                     "SpeculativePreconnectValidation")},
 
-#if defined(OS_ANDROID)
     {"enable-async-dns", flag_descriptions::kAsyncDnsName,
-     flag_descriptions::kAsyncDnsDescription, kOsAndroid,
+     flag_descriptions::kAsyncDnsDescription, kOsAll,
      FEATURE_VALUE_TYPE(features::kAsyncDns)},
-#endif  // defined(OS_ANDROID)
 
     {"enable-overflow-icons-for-media-controls",
      flag_descriptions::kOverflowIconsForMediaControlsName,
--- a/chrome/common/chrome_features.cc
+++ b/chrome/common/chrome_features.cc
@@ -77,7 +77,7 @@ const base::Feature kAssetDownloadSugges
 // Enables the built-in DNS resolver.
 const base::Feature kAsyncDns {
   "AsyncDns",
-#if defined(OS_CHROMEOS) || defined(OS_MACOSX) || defined(OS_ANDROID)
+#if defined(OS_CHROMEOS) || defined(OS_ANDROID)
       base::FEATURE_ENABLED_BY_DEFAULT
 #else
       base::FEATURE_DISABLED_BY_DEFAULT
